src/pudl_rmi/connect_deprish_to_ferc1.py

```diff
@@ -650,3 +652,141 @@ def _allocate_col(
+    assert(len(plant_bad) <= 53)
+    assert(len(util_bad) <= 88)
+
+    util_bad, plant_bad = data_col_test(
+        df1=df1, df2=df2, data_col='unaccrued_balance_w_common')
+
+    # known baddies. We need to track these down
+    assert(len(plant_bad) <= 35)
+    assert(len(util_bad) <= 70)
+
+
+def data_col_test(df1, df2, data_col: str) -> pd.DataFrame:
+    """
+    Check consistency of column at the plant and utility level in two inputs.
+
+    Args:
+        df1: One dataframe to sum and check consistency with ``df2``. Both
+            dfs should have columns ``data_col`` as well as identifying
+            columns: ``['report_date', 'data_source', 'utility_id_pudl',
+            'plant_id_eia']``
+        df2: Other dataframe to sum and check consistency against ``df1``.
+        data_col: data column to check. Column must be in both ``df1`` and
+            ``df2``.
+
+    """
+    util_test = gb_test(
+        df1,
+        df2,
+        data_col=data_col,
+        by=['report_year', 'data_source', 'utility_id_pudl']
+    )
+
+    logger.info(
+        f"Duke utility level data is consistent for {data_col}!! Go forth and "
+        "close them coal plants lil tables")
+
+    plant_test = gb_test(
+        df1,
+        df2,
+        data_col=data_col,
+        by=['report_year', 'data_source', 'utility_id_pudl', 'plant_id_eia']
+    )
+
+    util_bad = util_test[~util_test.match & util_test.match.notnull()]
+    plant_bad = plant_test[~plant_test.match & plant_test.match.notnull()]
+    logger.warn(
+        f"We have {len(util_bad)} utilities and {len(plant_bad)} "
+        f"plants who's {data_col} doesn't match the input."
+    )
+
+    # we know the duke utilities should be correct!
+    assert(util_test.loc[2018, 'FERC', 90].match)
+    assert(util_test.loc[2018, 'FERC', 97].match)
+    return util_bad, plant_bad
+
+
+def gb_test(
+    df1: pd.DataFrame,
+    df2: pd.DataFrame,
+    data_col: str,
+    by: List[str]
+) -> pd.DataFrame:
+    """
+    Merge two grouped input tables to determine if summed data column matches.
+
+    Args:
+        df1: One dataframe to sum and check consistency with ``df2``.
+        df2: Other dataframe to sum and check consistency against ``df1``.
+        data_col: data column to check. Column must be in both ``df1`` and
+            ``df2``.
+    """
+    return (
+        pd.merge(
+            _group_sum_col(
+                df1, data_col=data_col, by=by),
+            _group_sum_col(
+                df2, data_col=data_col, by=by),
+            right_index=True, left_index=True,
+            suffixes=('_1', '_2'),
+            how='outer'
+        )
+        .assign(
+            match=lambda x: np.where(
+                x[f"{data_col}_1"].notnull() & x[f"{data_col}_2"].notnull(),
+                np.isclose(x[f"{data_col}_1"], x[f"{data_col}_2"]),
+                pd.NA
+            ),
+            diff=lambda x: x[f"{data_col}_1"] / x[f"{data_col}_2"],
```

Comment:
`diff` to me says "difference" which means A - B, not A / B. Are you trying to get a sense of by how much it's off here? Instead you might want to calculate the equivalent `rtol` (relative tolerance) and `atol` (absolute tolerance) values that would be required for `np.isclose()` to return `True`.

Reply:
I like the strict default tolerances of `np.isclose` for this test. I'd like to keep them. Largely because these input values should be exactly the same, because they started the same and have just been through some processing.

I changed `diff` to `off_rate`. This feels good to me. I like to be able to see the bulk rate at which these values are off. It's useful to see if one value is say double or triple the other.

Reply:
I wasn't suggesting that you change the parameters for the `np.isclose()` check, just that the relative (fractional) and absolute (difference) tolerances that it uses are an already existing and clearly defined way of measuring how different two quantities are. `off_rate` still doesn't mean anything to me as a label (and I didn't understand what it was in the other part of the module where you're checking if it's null). Maybe `data_ratio` or `fractional_diff` where:
```py
fractional_diff=lambda x: 1.0 - x[f"{data_col}_1"] / x[f"{data_col}_2"]
```

Reply:
This may be a bad personal preference but I tend to like:

`data_ratio=lambda x: x[f"{data_col}_1"] / x[f"{data_col}_2"]`